lib/auth_plug.ex

```diff
@@ -26,66 +26,54 @@ defmodule AuthPlug do
-        #  so the check for JWT *before* anything else.
-        conn.query_string =~ "jwt" ->
-          query = URI.decode_query(conn.query_string)
-          Map.get(query, "jwt")
-
-        # Check for JWT in Headers:
-        Enum.count(get_req_header(conn, "authorization")) > 0 ->
-          conn.req_headers
-          |> List.keyfind("authorization", 0)
-          |> get_token_from_header()
-
-        #  Check for Person in Plug.Conn.assigns
-        Map.has_key?(conn.assigns, :jwt) && not is_nil(conn.assigns.jwt) ->
-          conn.assigns.jwt
-
-        # Check for Session in Plug.Session:
-        not is_nil(get_session(conn, :jwt)) ->
-          get_session(conn, :jwt)
-
-        # By default return nil so auth check fails
-        true ->
-          nil
-      end
 
-    validate_token(conn, jwt, options)
-  end
+    jwt = get_jwt(conn)
 
-  @doc """
-  `session_options/0` returns the list of Phoenix/Plug Session options.
-  This is useful if you need to check them or use them somewhere else.
-  """
-  def session_options() do
-    [
-      store: :cookie,
-      key: "_auth_key",
-      secret_key_base: System.get_env("SECRET_KEY_BASE"),
-      signing_salt: AuthPlug.Token.client_secret
-    ]
-  end
+    case AuthPlug.Token.verify_jwt(jwt) do
+      {:ok, values} ->
+        put_current_token(conn, jwt, values)
 
-  @doc """
-  `setup_session/1` configures the Phoenix/Plug Session.
-  """
-  def setup_session(conn) do
-    conn = put_in(conn.secret_key_base, System.get_env("SECRET_KEY_BASE"))
+      # log the JWT verify error then redirect:
+      {:error, reason} ->
+        Logger.error(Kernel.inspect(reason))
+        redirect_to_auth(conn, options)
+    end
+  end
 
-    opts = session_options() |> Plug.Session.init()
+  defp get_jwt(conn) do
+    cond do
+      # First Check for JWT in URL Query String.
+      # We want a *new* session to supercede any expired session,
+      #  so the check for JWT *before* anything else.
+      conn.query_string =~ "jwt" ->
+        query = URI.decode_query(conn.query_string)
+        Map.get(query, "jwt")
+
+      # Check for JWT in Headers:
+      Enum.count(get_req_header(conn, "authorization")) > 0 ->
+        conn.req_headers
+        |> List.keyfind("authorization", 0)
+        |> get_token_from_header()
+
+      #  Check for Person in Plug.Conn.assigns
+      Map.has_key?(conn.assigns, :jwt) && not is_nil(conn.assigns.jwt) ->
+        conn.assigns.jwt
+
+      # Check for Session in Plug.Session:
+      not is_nil(get_session(conn, :jwt)) ->
+        get_session(conn, :jwt)
+
+      # By default return nil so auth check fails
+      true ->
+        nil
+    end
+  end
 
-    conn
-    |> Plug.Session.call(opts)
-    |> fetch_session()
-    |> configure_session(renew: true)
+  def put_current_token(conn, jwt, values) do
+    # convert map of string to atom: stackoverflow.com/questions/31990134
+    claims = for {k, v} <- values, into: %{}, do: {String.to_atom(k), v}
```

Comment:
I've since discovered [`String.to_existing_atom/1`](https://hexdocs.pm/elixir/String.html#to_existing_atom/1)
which is better for memory management.